sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -1312,6 +1312,117 @@ class SumBinaryFixer(toType: DataType, isAnsi: Boolean)
   }
 }
 
+/**
+ * Fixes up a sum operation for unbounded preceding to unbounded following
+ * @param errorOnOverflow if we need to throw an exception when an overflow happens or not.
+ */
+class SumUnboundedToUnboundedFixer(resultType: DataType, failOnError: Boolean)
+    extends BatchedUnboundedToUnboundedWindowFixer {
+
+  private val OVERFLOW_MESSSAGE =
+    "One or more rows overflow for Add operation in SumUnboundedToUnboundedFixer"
+
+  private var previousValue: Option[Scalar] = None
+
+  override def updateState(scalar: Scalar): Unit = {
+    if (scalar.isValid) {
+      previousValue match {
+        case None =>
+          previousValue = Some(scalar.incRefCount())
+        case Some(prev) =>
+          withResource(prev) { _ =>
+            prev.getType.getTypeId match {
+              case DType.DTypeEnum.INT8 =>
```

Comment:
I thought that INT64 was the smallest sum value supported?

https://github.com/apache/spark/blob/93af0848e467fe4d58c0fb1242b738931390d6f8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala#L74-L81

I thought for Sum that would either in the logical plan, where it would cast the input to the intermediate type before doing the agg, or for newer versions of Spark it would just happen in SUM itself as a part of getting ready to do the agg.

I think that just means we can drop INT8, INT16, and INT32 code from this section and throw an error if we see them.